src/java/map/FloatMask.java

```diff
@@ -572,6 +572,64 @@ public FloatMask removeAreasOfSpecifiedSizeWithLocalMaximums(int minSize, int ma
         return this;
     }
 
+    public FloatMask scaleFloatMaskBy(float scaleMultiplier) {
+        for (int x = 0; x < getSize(); x++) {
+            for (int y = 0; y < getSize(); y++) {
+                multiply(x, y, scaleMultiplier);
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask scaleFloatMaskTo(float newFloatMax) {
+        float scaleMultiplier = newFloatMax / getMax();
+        for (int x = 0; x < getSize(); x++) {
+            for (int y = 0; y < getSize(); y++) {
+                multiply(x, y, scaleMultiplier);
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask scaleFloatMaskFromTo(float oldFloatMax, float newFloatMax) {
+        float scaleMultiplier = newFloatMax / oldFloatMax;
+        for (int x = 0; x < getSize(); x++) {
+            for (int y = 0; y < getSize(); y++) {
+                multiply(x, y, scaleMultiplier);
+            }
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public int getRankOfPoint(int x, int y) {
+        int rank = 1;
+        float value = get(x, y);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                if(value > get(a, b) || (value == get(a, b) && x * 100 + y > a * 100 + b)) {
```

Comment:
this will result in a lot of ranks due to it being float values. 

also what is the 100 for?

Reply:
generic large multiplier to make it so it prioritizes x-values before y-values for tie-breaking (I did this as an alternative to more if-statements for the same functionality)

Reply:
would it be better to make the multiplier the size? Same effect but is also related to the mask